src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/Metrics/IomtMetrics.cs

```diff
@@ -0,0 +1,278 @@
+           });
+
+        /// <summary>
+        /// The latency between event ingestion and output to FHIR processor.
+        /// </summary>
+        public static Metric MeasurementIngestionLatency()
+        {
+            return _measurementIngestionLatency;
+        }
+
+        /// <summary>
+        /// The number of measurement groups generated by the FHIR processor based on provided input.
+        /// </summary>
+        public static Metric MeasurementGroup()
+        {
+            return _measurementGroup;
+        }
+
+        /// <summary>
+        /// The number of measurement readings to import to FHIR.
+        /// </summary>
+        public static Metric Measurement()
+        {
+            return _measurement;
+        }
+
+        /// <summary>
+        /// The number of input events received.
+        /// </summary>
+        public static Metric DeviceEvent()
+        {
+            return _deviceEvent;
+        }
+
+        /// <summary>
+        /// The number of normalized events generated for further processing.
+        /// </summary>
+        public static Metric NormalizedEvent()
+        {
+            return _normalizedEvent;
+        }
+
+        /// <summary>
+        /// The latency between the event ingestion time and normalization processing. An increase here indicates a backlog of messages to process.
+        /// </summary>
+        public static Metric DeviceEventProcessingLatency()
+        {
+            return _deviceEventProcessingLatency;
+        }
+
+        /// <summary>
+        /// An exception thrown when the patient and device references and resources do not match.
+        /// </summary>
+        public static Metric PatientDeviceMismatchException()
+        {
+            return _patientDeviceMismatch;
+        }
+
+        /// <summary>
+        /// An exception thrown when the FHIR resource does not support the provided type as a value.
+        /// </summary>
+        public static Metric NotSupportedException()
+        {
+            return _notSupportedException;
+        }
+
+        /// <summary>
+        /// Multiple FHIR resources were found when only one was expected.
+        /// </summary>
+        public static Metric MultipleResourceFoundException()
+        {
+            return _multipleResourceFoundException;
+        }
+
+        /// <summary>
+        /// The mapping template is not defined.
+        /// </summary>
+        public static Metric TemplateNotFoundException()
+        {
+            return _templateNotFoundException;
+        }
+
+        /// <summary>
+        /// An exception recorded when grouping correlation id but the correlation id is null or not found.
+        /// </summary>
+        public static Metric CorrelationIdNotDefinedException()
+        {
+            return _correlationIdNotDefinedException;
+        }
+
+        public static Metric FhirResourceNotFoundException(ResourceType resourceType)
```

Comment:
Don't love this here.  Feel like it should be part of the exception definition and resolved through the interface you added or handled via an ExtensionMethod

Reply:
Updated